pkg/scheduler/api/v1/types.go

```diff
@@ -164,6 +167,47 @@ type ExtenderArgs struct {
 	NodeNames *[]string `json:"nodenames,omitempty"`
 }
 
+// ExtenderPreemptionResult represents the result returned by preemption phase of extender.
+// Only set NodeNameToMetaVictims if ExtenderConfig.NodeCacheCapable == true. Otherwise, only set NodeToVictims.
+type ExtenderPreemptionResult struct {
+	NodeNameToMetaVictims map[string]*MetaVictims  `json:"nodeNameToMetaVictims,omitempty"`
+	NodeToVictims         map[*apiv1.Node]*Victims `json:"nodeToVictims,omitempty"`
+}
+
+// ExtenderPreemptionArgs represents the arguments needed by the extender to preempt pods on nodes.
+type ExtenderPreemptionArgs struct {
+	// Pod being scheduled
+	Pod *apiv1.Pod `json:"pod"`
+	// Victims map generated by scheduler preemption phase
+	// Only set NodeNameToMetaVictims if ExtenderConfig.NodeCacheCapable == true. Otherwise, only set NodeToVictims.
+	NodeToVictims         map[*apiv1.Node]*Victims `json:"nodeToVictims,omitempty"`
+	NodeNameToMetaVictims map[string]*MetaVictims  `json:"nodeNameToMetaVictims,omitempty"`
+}
+
+// Victims represents:
+//   pods:  a group of pods expected to be preempted.
+//   numPDBViolations: the count of violations of PodDisruptionBudget
+type Victims struct {
+	Pods             []*apiv1.Pod `json:"pods"`
+	NumPDBViolations int          `json:"numPDBViolations"`
+}
+
+// MetaPod represent identifier for a v1.Pod
+type MetaPod struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+	UID       string `json:"uid"`
```

Comment:
I believe UID should be enough. Do we need to set name and namespace as well?

Reply:
Just back from vacation :) Yes UID should be enough.